src/extensions/default/CSSAtRuleCodeHints/main.js

```diff
@@ -0,0 +1,117 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    // Load dependent modules
+    var AppInit         = brackets.getModule("utils/AppInit"),
+        CodeHintManager = brackets.getModule("editor/CodeHintManager"),
+        AtRulesText     = require("text!AtRulesDef.json"),
+        AtRules         = JSON.parse(AtRulesText);
+
+
+    /**
+     * @constructor
+     */
+    function AtRuleHints() {
+    }
+
+    // As we are only going to provide @rules name hints
+    // we should claim that we don't have hints for anything else
+    AtRuleHints.prototype.hasHints = function (editor, implicitChar) {
+        var pos = editor.getCursorPos(),
+            token = editor._codeMirror.getTokenAt(pos),
+            cmState;
+
+        this.editor = editor;
+
+        if (token.state.base && token.state.base.localState) {
+            cmState = token.state.base.localState;
+        } else {
+            cmState = token.state;
+        }
+
+        // Check if we are at '@' rule 'def' context
+        if ((token.type === "def" && cmState.context.type === "at")
+                || (token.type === "variable-2" && (cmState.context.type === "top" || cmState.context.type === "block"))) {
+            this.filter = token.string;
+            return true;
+        } else {
+            this.filter = null;
+            return false;
+        }
+    };
+
+    AtRuleHints.prototype.getHints = function (implicitChar) {
+        var pos     = this.editor.getCursorPos(),
+            token   = this.editor._codeMirror.getTokenAt(pos);
+
+        this.filter = token.string;
+        this.token = token;
+        
+        if (!this.filter) {
+            return null;
+        }
+
+        // Filter the property list based on the token string
+        var result = Object.keys(AtRules).map(function (key) {
```

Comment:
I might be wrong, but I think `filter` would be more appropriate than `map` for the first loop. This would also make the following `filter` loop unnecessary.